sdk/pubkey/src/lib.rs

```diff
@@ -36,46 +39,50 @@ use pinocchio::syscalls::sol_sha256;
 /// (off-curve) program derived address. It is intended for use in cases where the
 /// seeds, bump, and program id are known to be valid, and the caller wants to derive
 /// the address without incurring the cost of the `create_program_address` syscall.
-pub fn derive_address<const N: usize>(seeds: &[&[u8]; N], bump: u8, program_id: &Pubkey) -> Pubkey {
+pub fn derive_address<const N: usize>(
+    seeds: &[&[u8]; N],
+    bump: Option<u8>,
+    program_id: &Pubkey,
+) -> Pubkey {
     const {
-        assert!(
-            N <= MAX_SEEDS,
-            "number of seeds must be less than MAX_SEEDS"
-        );
+        assert!(N < MAX_SEEDS, "number of seeds must be less than MAX_SEEDS");
     }
 
     const UNINIT: MaybeUninit<&[u8]> = MaybeUninit::<&[u8]>::uninit();
-    let mut data = [UNINIT; MAX_SEEDS + 3];
+    let mut data = [UNINIT; MAX_SEEDS + 2];
```

Comment:
Why reduce the maximum number of allowed seeds here?
Was it incorrect in the version with `bump`?

Reply:
Yeah, the `bump` counts as a seed.